### PR TITLE
Validation: add Inverted Result

### PR DESCRIPTION
**Pull request by hhunderter**
# Description

Fügt in folgenden Abfragen eine Invertieren-Funktion hinzu:
- required
- unique
- numeric
- integer

Mit z.b. 'NOTrequired' muss das Feld Leer sein

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# This PR has been tested in the following browsers:

- [ ] Chrome
- [X] Firefox
- [ ] Opera
- [ ] Edge
- [ ] IE

### CODE REVIEW

File: application/libraries/Ilch/Validation/Validators/Base.php [hunk 1]
```diff
@@ -199,14 +224,38 @@ public function setParameters($parameters)
         return $this;
     }
 
+    /**
+     * Get the value of Invert Result for this validator.
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Base.php:228]: ``` * Get the value of invertResult for this validator. ```

File: application/libraries/Ilch/Validation/Validators/Base.php [hunk 2]
```diff
@@ -199,14 +224,38 @@ public function setParameters($parameters)
...
+     *
+     * @return bool
+     */
+    public function getInvertResult()
+    {
+        return $this->invertResult;
+    }
+
+    /**
+     * Set the value of Invert Result for this validator.
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Base.php:238]: ```* Set the value of invertResult for this validator.```

File: application/libraries/Ilch/Validation/Validators/Integer.php [hunk 3]
```diff
@@ -17,6 +17,20 @@ class Integer extends Base
      */
     protected $errorKey = 'validation.errors.integer.mustBeInteger';
 
+    /**
+     * defines whether logic can be negated .
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Integer.php:21]: ```* Defines whether logic can be negated.```

File: application/libraries/Ilch/Validation/Validators/Numeric.php [hunk 4]
```diff
@@ -17,6 +17,20 @@ class Numeric extends Base
      */
     protected $errorKey = 'validation.errors.numeric.mustBeNumeric';
 
+    /**
+     * defines whether logic can be negated .
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Numeric.php:21]: ```* Defines whether logic can be negated.```

File: application/libraries/Ilch/Validation/Validators/Required.php [hunk 5]
```diff
@@ -17,6 +17,20 @@ class Required extends Base
      */
     protected $errorKey = 'validation.errors.required.fieldIsRequired';
 
+    /**
+     * defines whether logic can be negated .
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Required.php:21]: ```* Defines whether logic can be negated.```

File: application/libraries/Ilch/Validation/Validators/Same.php [hunk 6]
```diff
@@ -17,6 +17,20 @@ class Same extends Base
      */
     protected $errorKey = 'validation.errors.same.fieldsDontMatch';
 
+    /**
+     * defines whether logic can be negated .
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Same.php:21]: ```* Defines whether logic can be negated.```

File: application/libraries/Ilch/Validation/Validators/Base.php [hunk 7]
```diff
@@ -247,6 +296,30 @@ public function setErrorKey($errorKey)
         return $this;
     }
 
+    /**
+     * Get the value of inverted Error Key.
```
Comment (blackcoder87) [inline application/libraries/Ilch/Validation/Validators/Base.php:300]: invert error key or invertErrorKey ?? Not really conistent?

Comment (hhunderter) [reply application/libraries/Ilch/Validation/Validators/Base.php:300]: Wie ist das gemeint?

**Pull request merged**